monitored-httpclient/src/main/java/com/github/nhenneaux/resilienthttpclient/monitoredclientpool/SingleIpHttpClient.java

```diff
@@ -116,17 +119,44 @@ void checkHealthStatus() {
                     .join();
 
             LOGGER.log(Level.INFO, () -> "Checked health for URI " + healthUri + ", status is `" + statusCode + "`" + timingLogStatement(start));
-            healthy.set(statusCode >= 200 && statusCode <= 499);
+
+            refreshCounterWithStatusCode(statusCode);
         } catch (RuntimeException e) {
             LOGGER.log(Level.WARNING, () -> "Failed to check health for address " + healthUri + ", error is `" + e + "`" + timingLogStatement(start), e);
-            healthy.set(false);
+
+            refreshCounterWithException();
         }
     }
 
+    /**
+     * Validates if the client is within failed response count threshold.
+     */
+    boolean isWithinFailedResponseCountThreshold() {
+        final int failureResponseCountThreshold = serverConfiguration.getFailureResponseCountThreshold();
+
+        return failureResponseCountThreshold == -1 || httpRequestStatusCounter.getFailureResponseCount() < failureResponseCountThreshold;
+    }
+
+    void refreshCounterWithStatusCode(final int statusCode) {
+        final boolean successCode = isSuccessCode(statusCode);
+
+        healthy.set(successCode);
+        httpRequestStatusCounter.count(successCode);
+    }
+
+    void refreshCounterWithException() {
+        healthy.set(false);
```

Comment:
it is another change I would prefer to split the change with the failure threshold since it could lad to quite unstable behavior

Reply:
Done. No health update during counter update with exception